python/ray/serve/metric/sink.py

```diff
@@ -0,0 +1,139 @@
+from typing import Dict
+from collections import Counter, namedtuple
+
+import ray
+from ray.serve.metric.types import MetricType, MetricMetadata, MetricBatch
+from ray.serve.utils import _get_logger
+
+logger = _get_logger()
+
+
+def make_metric_namedtuple(metadata: MetricMetadata, record: MetricBatch):
+    fields = ["name", "type"]
+    fields += list(metadata.default_labels.keys())
+    fields += list(record.labels.keys())
+
+    merged_labels = metadata.default_labels.copy()
+    merged_labels.update(record.labels)
+
+    tuple_type = namedtuple(metadata.name, fields)
+    return tuple_type(name=metadata.name, type=metadata.type, **merged_labels)
+
+
+class BaseSink:
```

Comment:
I don't really like the name `BaseSink` because it doesn't communicate that this is actually to be used as a client (either pull- or push-based). How about something like `MetricsExporter`?